### DISCUSSION TITLE
Orleans in Kubernetes with external client.

### DISCUSSION THREAD

**Discussion by Rafajax95** [dotnet/orleans#8164] [Q&A]
Hello!
I have a case which, I believe, isn't fully covered in documentation.
https://learn.microsoft.com/en-us/dotnet/orleans/deployment/kubernetes
https://learn.microsoft.com/en-us/dotnet/orleans/host/configuration-guide/typical-configurations

I want to host cluster of silos in Kubernetes and I want to have a client of the cluster hosted externally, let's say Azure Function
Now I have a problem. The external client cannot reach silos because the ip address is an internal pod address.
How should I configure client and silos to achieve that goal, ideally with usage of Ingress?
Is this a recommended approach at all?

I observe the community from some time and see really exiting ideas so I thought that maybe someone have experience with that and can help me a bit to find a proper solution :)

**Comment by EPinci** [comment]
Hi,

if you're running your cluster with Kubernetes, you might deploy it with Azure CNI and attaching the Azure Function to the same vnet you could reach the pods directly, somehow fixing your scenario. I would still call this a somewhat weird scenario.

If the intent is to "hide" the cluster implementation, since you can "proxy" the silos with an ingress, you might end up needing to write your own rest proxy that, in turn, you can expose with an ingress.

HTH.

**Reply by Rafajax95** [reply]
Thanks for response.
At the end we decided to co-host Orleans client together with server in the same application. We figured out that this is more recommended Orleans usage for many reasons.
